iocore/utils/PartitionedMap.h

```diff
@@ -0,0 +1,287 @@
+#ifndef _PartitionedMap_h
+#define _PartitionedMap_h
+
+#include "stdint.h"
+#include <cstddef>
+#include <mutex>
+#include <unordered_map>
+#include <atomic>
+#include <thread>
+
+#include "ts/ink_assert.h"
+
+using Mutex     = std::recursive_mutex;
```

Comment:
This seems problematic given the wide of use `Mutex` in other core header files.

Reply:
I'm moving this into each class that needs it.